starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,234 @@
+                mTextView5.getText() == mTextView9.getText() && mTextView1.getText() != "") {
+            win = true;
+        } else if (mTextView3.getText() == mTextView5.getText() &&
+                mTextView5.getText() == mTextView7.getText() && mTextView3.getText() != "") {
+            win = true;
+        } else {
+            win = false;
+        }
+    }
 
-public class GameActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_game);
+
+        Intent intent = getIntent();
+        mUserName1 = intent.getStringExtra("Username1");
+        mUsername2 = intent.getStringExtra("Username2");
+
+        mTextView1 = (TextView) findViewById(R.id.textView1);
+        mTextView2 = (TextView) findViewById(R.id.textView2);
+        mTextView3 = (TextView) findViewById(R.id.textView3);
+        mTextView4 = (TextView) findViewById(R.id.textView4);
+        mTextView5 = (TextView) findViewById(R.id.textView5);
+        mTextView6 = (TextView) findViewById(R.id.textView6);
+        mTextView7 = (TextView) findViewById(R.id.textView7);
+        mTextView8 = (TextView) findViewById(R.id.textView8);
+        mTextView9 = (TextView) findViewById(R.id.textView9);
+        mGameMessage = (TextView) findViewById(R.id.game_message_text);
+        mBackButton = (Button) findViewById(R.id.backbutton);
+
+        mGameMessage.setText(mUserName1 +"'s Turn");
+
+        mTextView1.setOnClickListener(this);
+        mTextView2.setOnClickListener(this);
+        mTextView3.setOnClickListener(this);
+        mTextView4.setOnClickListener(this);
+        mTextView5.setOnClickListener(this);
+        mTextView6.setOnClickListener(this);
+        mTextView7.setOnClickListener(this);
+        mTextView8.setOnClickListener(this);
+        mTextView9.setOnClickListener(this);
+
+        mBackButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent resultIntent = new Intent();
+                resultIntent.putExtra("Winner", mWinner);
+                setResult(RESULT_OK,resultIntent);
+                finish();
+            }
+        });
+
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (win == false) {
+        switch (view.getId()) {
+                case R.id.textView1:
+                    if (mTextView1.getText() == "") {
+                        mTurn++;
+                        whoseTurn();
+                        mTextView1.setText(mLetter);
+                        timer.start();
+                    }
+                    break;
+                case R.id.textView2:
+                    if (mTextView2.getText() == "") {
+                        mTurn++;
+                        whoseTurn();
+                        mTextView2.setText(mLetter);
+                        timer.start();
```

Comment:
Hey Charlie,
To be honest; I also thought of that; but for every case; I had to put in a different TextView object and I couldn't quite figure out how. Can you write a simple example method for me that would take into account the changing TextViews?
Thank you.